### DISCUSSION TITLE
EMBA v1.3.0 - AI-Assisted Firmware Analysis

### DISCUSSION THREAD

**Discussion by m-1-k-3** [e-m-b-a/emba#720] [Announcements]
_Q: Can we use AI for firmware analysis?_
A: Sure, let's do it! EMBA now supports **AI-assisted firmware analysis**. 

<p align="center">
<img src=https://github.com/e-m-b-a/emba/assets/497520/ede45fb6-960a-43ca-beaa-f0fb28345995 width="700">
</p>

Again, we rise the bar in the field of Open-Source firmware security analysis. After establishing [user-mode emulation](https://github.com/e-m-b-a/emba/wiki/User-mode-Emulator) or [system emulation](https://github.com/e-m-b-a/emba/wiki/System-emulation) this time we moved to AI-assisted firmware analysis. More details about our AI integration are available in our [Wiki](https://github.com/e-m-b-a/emba/wiki/AI-supported-firmware-analysis)

***

**#Hackersummercamp ahead!**
We got the amazing opportunity to show _EMBA_ at the BSides conference in Las Vegas. The schedule is available [here](https://bsideslv.org/talks#DWYRJ3).

Additionally, you will find us with a live _EMBA_ demo at [Black Hat Arsenal](https://www.blackhat.com/us-23/arsenal/schedule/index.html#emba--from-firmware-to-exploit-31537)

_See you all in Vegas_

***

Beside your ongoing support with feedback, testing, working on issues and spreading _EMBA_ you can now also support _EMBA_ as a sponsor.

![image](https://user-images.githubusercontent.com/497520/213381032-3373ae80-ca0d-4b55-a28d-1fed48e62a18.png)

Check it out [here](https://github.com/sponsors/e-m-b-a) and start being an essential part of the future of _EMBA_

***

## What's Changed
* Exit of add_partition in L10 by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/430
* log dir on dep check by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/428
* Nikto dep fix by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/429
* cwe-checker install latest master by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/431
* Further trickest blacklist entries by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/432
* Freetzng-fix by @BenediktMKuehne in https://github.com/e-m-b-a/emba/pull/433
* update sub-shell pwd fix by @BenediktMKuehne in https://github.com/e-m-b-a/emba/pull/435
* Add Packetstorm and Snyk PoC sources by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/434
* Full install fixes by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/436
* s115 - empty log handling by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/438
* Minimal cve-search installation / Dependency issues by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/442
* blacklist update by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/441
* Introducing module_wait helper function by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/439
* Fix dependencies by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/445
* Code cleanup - comments by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/446
* Copyright updates 2023 by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/447
* Kernel downloader and vulnerability verifier by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/451
* cron job fix by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/453
* L10 improvements, more services by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/454
* Kernel config analysis by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/455
* Update the known exploit behaviour by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/458
* example disable profile by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/457
* Refactoring by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/462
* exploit databases updated by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/466
* S12 - checksec implementation fix by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/463
* Improve stop of system emulation by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/465
* Hexagon support by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/467
* Lighttpd analysis module by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/469
* s08 safe_echo fix by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/470
* p35 - true to not fail, s26 - check for files by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/471
* JTR crack multiple hash types by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/473
* deprecated -l option by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/476
* s36 fixes, renamed p61 by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/477
* System emulator improvements by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/478
* Respect module blacklist in waiting state / Installer fix by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/479
* Exploit database update, debug mode, command line tests by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/481
* Add wordlist mechanism to s109 by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/482
* csv export of p59, p60 and p70 by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/483
* disk space monitor, rpm package analysis by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/485
* Improve output of help command by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/492
* Setup further workflows by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/490
* Remove timezone setting by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/494
* Refactor, PID log, Github actions, APKHunt by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/495
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/498
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/497
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/496
* Improve restart EMBA analysis feature by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/499
* Fix install with pip v23+ by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/500
* Another PIPv23 fix by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/501
* return if empty by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/502
* Input validation by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/505
* Check for update setting by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/504
* Routersploit update workflow by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/503
* Dependency checker, workflow by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/506
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/509
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/510
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/512
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/514
* System emulation improvements, workflow by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/515
* CVE state message printing by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/518
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/528
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/527
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/525
* Routersploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/524
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/523
* Trickest PoC database update by @github-actions in https://github.com/e-m-b-a/emba/pull/526
* Input adjustment by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/529
* version validation by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/530
* PATH variable bug by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/531
* EMBA v1.2.2 - Blue Hat edt. by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/532
* Sponsoring issues by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/534
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/536
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/539
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/537
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/540
* L25 improvements / multiple little fixes by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/535
* L10 module improvements by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/543
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/545
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/547
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/548
* New version strings (Flex and NBTscan) by @HoxhaEndri in https://github.com/e-m-b-a/emba/pull/549
* L10 improvement round x by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/550
* links in templates by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/555
* Freetz extraction module deprecated by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/554
* fix for #551 by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/553
* Testing workflows by @BenediktMKuehne in https://github.com/e-m-b-a/emba/pull/541
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/563
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/562
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/560
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/559
* Improve web crawler (L25) by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/557
* Updated installer.sh for "ubuntu debian" /etc/os-release and new version string by @HoxhaEndri in https://github.com/e-m-b-a/emba/pull/552
* SNMP module improvements by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/565
* Remove warning apt-key is deprecated by @HoxhaEndri in https://github.com/e-m-b-a/emba/pull/564
* update entropy output by @BenediktMKuehne in https://github.com/e-m-b-a/emba/pull/566
* Ignore files containing the following paths: */dev/* */proc/* */sys/* by @HoxhaEndri in https://github.com/e-m-b-a/emba/pull/569
* Fix arch detection in f50 by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/567
* Install fixes by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/570
* fix l10 error case by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/571
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/573
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/575
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/576
* Trickest PoC database update by @github-actions in https://github.com/e-m-b-a/emba/pull/574
* Improved default profile handling / running modules script by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/572
* Fail fetch aspnetcore-targeting-pack when cleaning up by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/579
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/581
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/582
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/585
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/584
* Trickest PoC database update by @github-actions in https://github.com/e-m-b-a/emba/pull/583
* fix actions, fix l10 lnk fixer by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/580
* remove unneeded resource by @BenediktMKuehne in https://github.com/e-m-b-a/emba/pull/586
* Revert "remove unneeded resource" by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/587
* SBOM generation fix for non vuln components by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/589
* Avoiding /proc and /sys paths (-xdev) in symlink script and check for missing symlinks in s115 by @HoxhaEndri in https://github.com/e-m-b-a/emba/pull/590
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/597
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/596
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/594
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/593
* Lua script analysis support, UPnP live module, improvements by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/591
* R.I.P. binwalk by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/598
* ignore named pipe by @HoxhaEndri in https://github.com/e-m-b-a/emba/pull/601
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/607
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/606
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/604
* apk extraction fix by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/603
* R2 decompiler integration by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/608
* url update for sasquatch deb by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/609
* update ubuntu libssl source by @BenediktMKuehne in https://github.com/e-m-b-a/emba/pull/610
* Small cleanup fixes by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/611
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/616
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/615
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/614
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/613
* Hnap detection support for system emulator by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/612
* Version 1.2.3 by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/621
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/626
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/625
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/623
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/624
* Multiple bug fixes by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/629
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/633
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/632
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/630
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/631
* Restart emulation improvements by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/634
* Fix all the things by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/636
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/639
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/642
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/643
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/640
* Trickest PoC database update by @github-actions in https://github.com/e-m-b-a/emba/pull/641
* Routersploit/emulation cleanup/installer by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/646
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/649
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/648
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/652
* Trickest PoC database update by @github-actions in https://github.com/e-m-b-a/emba/pull/650
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/651
* zombie handling by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/647
* JNAP detection by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/654
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/655
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/656
* Trickest PoC database update by @github-actions in https://github.com/e-m-b-a/emba/pull/657
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/659
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/658
* use env variable CONTAINER to choose the docker container by @HoxhaEndri in https://github.com/e-m-b-a/emba/pull/644
* JNAP unauth check by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/660
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/665
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/664
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/662
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/661
* trickest bl update by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/667
* Update workflows by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/668
* Adjust workflows / CWE checker update to v0.7 by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/669
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/671
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/670
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/674
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/673
* trickest blacklist update by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/675
* Trickest PoC database update by @github-actions in https://github.com/e-m-b-a/emba/pull/672
* s22 tear down functionality by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/677
* workflow, readme by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/678
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/679
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/681
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/682
* Trickest PoC database update by @github-actions in https://github.com/e-m-b-a/emba/pull/680
* Little cleanups by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/683
* Extended version string for proftpd by @HoxhaEndri in https://github.com/e-m-b-a/emba/pull/684
* typo, csv cleanup by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/687
* trickest blacklist update by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/688
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/690
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/691
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/693
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/694
* Trickest PoC database update by @github-actions in https://github.com/e-m-b-a/emba/pull/692
* Fix L15 report by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/695
* Improve web crawler by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/696
* Bug fixes by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/698
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/700
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/699
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/703
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/702
* Trickest PoC database update by @github-actions in https://github.com/e-m-b-a/emba/pull/701
* make qnap check strict mode compatible by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/704
* P60 - lost functions by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/705
* Semgrep rule disable by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/706
* Move to pip env for python installation by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/707
* Sh eval checks by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/710
* CISA known exploited database update by @github-actions in https://github.com/e-m-b-a/emba/pull/712
* Metasploit database update by @github-actions in https://github.com/e-m-b-a/emba/pull/711
* Snyk database update by @github-actions in https://github.com/e-m-b-a/emba/pull/714
* Packetstorm database update by @github-actions in https://github.com/e-m-b-a/emba/pull/715
* Trickest PoC database update by @github-actions in https://github.com/e-m-b-a/emba/pull/713
* EMBA ChatGPT integration by @BenediktMKuehne in https://github.com/e-m-b-a/emba/pull/602
* load python environment in updater by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/718
* GPT rating improvements by @m-1-k-3 in https://github.com/e-m-b-a/emba/pull/717
* Release 1.3.0 by @BenediktMKuehne in https://github.com/e-m-b-a/emba/pull/719

## New Contributors
* @github-actions made their first contribution in https://github.com/e-m-b-a/emba/pull/498
* @HoxhaEndri made their first contribution in https://github.com/e-m-b-a/emba/pull/549

**Full Changelog**: https://github.com/e-m-b-a/emba/compare/1.2.0-London-Calling...1.3.0-AI-for-EMBA

<hr /><em>This discussion was created from the release <a href='https://github.com/e-m-b-a/emba/releases/tag/1.3.0-AI-for-EMBA'>EMBA v1.3.0 - AI-Assisted Firmware Analysis</a>.</em>

**Comment by BenediktMKuehne** [comment]
That's a big one 🤩
Great work guys!
